### PR TITLE
Implement test autocompletion for bash/zsh/fish

### PR DESCRIPTION
**Pull request by tanishiking**
This PR enables

- bloop server to respond tests FQCN in a project to a testsfqcn autocomplete request like `bloop autocomplete --format <shell> --mode testsfqcn --project <project>`
- bash/zsh/fish to autocomplete test names after `bloop test <project> --only`

<details>
<summary>Here are demos</summary>

### zsh
[asciicast](https://asciinema.org/a/NaRcg4vNckhD2pYrvmo114hhi)

(in this video, autocompletion transformed the command from `bloop` to `blobloop`, but this is not actually happening..., I think this is a problem of `asciinema` and my zsh configuration...)

### bash
[asciicast](https://asciinema.org/a/89AAwTmMzW9gbEILMXnCtBveZ)

### fish
[asciicast](https://asciinema.org/a/rhypFT5yCDGMk5lurkot1zuGh)

</details>

---

test is failing, but I'm not sure how the test related to this PR.

### CODE REVIEW

**Comment by tanishiking** [comment]
Oops, some tests are failing, I'll fix them...

File: frontend/src/main/scala/bloop/engine/Interpreter.scala [hunk 1]
```diff
@@ -275,10 +275,13 @@ object Interpreter {
           completion <- cmd.format.showMainName(main)
         } state.logger.info(completion)
       case Mode.TestsFQCN =>
+        import ExecutionContext.scheduler
```
Comment (tanishiking) [inline frontend/src/main/scala/bloop/engine/Interpreter.scala]: I'm not sure which scheduler should we use here...

File: etc/fish/bloop.fish [hunk 2]
```diff
@@ -57,6 +57,23 @@ function __fish_seen_subcommand_from_project_commands
     __fish_seen_subcommand_from $prj
 end
 
+# https://github.com/fish-shell/fish-shell/blob/04795eb8ea63e9b68bc85342aae81b9a60cd09c8/share/functions/__fish_prev_arg_in.fish
+function __fish_prev_arg_in
```
[the remark this answers is not in the document]
Comment (tanishiking) [reply etc/fish/bloop.fish]: oops, fish-shell is GPL licensed...

**Comment by jvican** [comment]
Nice! @tanishiking Will have a look at this next Monday since i'm in vacation for some days, thanks for the new tests :smile:

**Pull request merged**